dll/win32/shell32/CDefView.cpp

```diff
@@ -783,40 +865,243 @@ void CDefView::UpdateListColors()
+{
+    // This function is only called during column management, performance is not critical.
+    for (UINT i = 0;; ++i)
+    {
+        HRESULT r = SHGetLVColumnSubItem(m_ListView.m_hWnd, i);
+        if ((UINT) r == FoldCol)
+            return i;
+        else if (FAILED(r))
+            return r;
+    }
+}
+
+HRESULT CDefView::MapListColumnToFolderColumn(UINT ListCol)
+{
+    // This function is called every time a LVITEM::iSubItem is mapped to
+    // a folder column (calls to GetDetailsOf etc.) and should be fast.
+    if (m_ListToFolderColMap)
+    {
+        UINT count = DPA_GetPtrCount(m_ListToFolderColMap);
+        if (ListCol < count)
+        {
+            HRESULT col = (INT)(INT_PTR) DPA_FastGetPtr(m_ListToFolderColMap, ListCol);
+            assert(col >= 0 && col == SHGetLVColumnSubItem(m_ListView.m_hWnd, ListCol));
+            return col;
+        }
+        else if (count)
+        {
+            TRACE("m_ListToFolderColMap cache miss while mapping %d\n", ListCol);
+        }
+    }
+    return SHGetLVColumnSubItem(m_ListView.m_hWnd, ListCol);
+}
+
+HRESULT CDefView::GetDetailsByFolderColumn(PCUITEMID_CHILD pidl, UINT FoldCol, SHELLDETAILS &sd)
+{
+    // According to learn.microsoft.com/en-us/windows/win32/shell/sfvm-getdetailsof
+    // the query order is IShellFolder2, IShellDetails, SFVM_GETDETAILSOF.
+    HRESULT hr;
     if (m_pSF2Parent)
     {
-        for (int i = 0; 1; i++)
+        hr = m_pSF2Parent->GetDetailsOf(pidl, FoldCol, &sd);
+    }
+    if (FAILED(hr) && m_pSDParent)
+    {
+        hr = m_pSDParent->GetDetailsOf(pidl, FoldCol, &sd);
+    }
+#if 0 // TODO
+    if (FAILED(hr))
+    {
+        FIXME("Try SFVM_GETDETAILSOF\n");
+    }
+#endif
+    return hr;
+}
+
+HRESULT CDefView::GetDetailsByListColumn(PCUITEMID_CHILD pidl, UINT ListCol, SHELLDETAILS &sd)
+{
+    HRESULT hr = MapListColumnToFolderColumn(ListCol);
+    if (SUCCEEDED(hr))
+        return GetDetailsByFolderColumn(pidl, hr, sd);
+    ERR("Unable to determine folder column from list column %d\n", (int) ListCol);
+    return hr;
+}
+
+HRESULT CDefView::LoadColumn(UINT FoldCol, UINT ListCol, BOOL Insert)
+{
+    WCHAR buf[MAX_PATH];
+    SHELLDETAILS sd;
+    HRESULT hr;
+
+    sd.str.uType = !STRRET_WSTR; // Protect ourself from broken implementations
+    hr = GetDetailsByFolderColumn(NULL, FoldCol, sd);
+    if (FAILED(hr))
+        return hr;
+    hr = StrRetToStrNW(buf, _countof(buf), &sd.str, NULL);
+    if (FAILED(hr))
+        return hr;
+
+    UINT chavewidth = CalculateCharWidth(m_ListView.m_hWnd);
+    if (!chavewidth)
+        chavewidth = 6; // 6 is a reasonable default fallback
+
+    LVCOLUMN lvc;
+    lvc.mask = LVCF_TEXT | LVCF_FMT | LVCF_WIDTH | LVCF_SUBITEM;
+    lvc.pszText = buf;
+    lvc.fmt = sd.fmt;
+    lvc.cx = sd.cxChar * chavewidth; // FIXME: DPI?
+    lvc.iSubItem = FoldCol; // Used by MapFolderColumnToListColumn & MapListColumnToFolderColumn
+    if ((int) ListCol == -1)
```

Comment:
```suggestion
    if ((int)ListCol == -1)
```
It helps seeing that the cast is applied first, before the == is evaluated. Nit: Actually all spaces after casts are questionable, but this one here hurts the readers interpretation most. Otherwise your PR lgtm.